manifests/010-prometheus-rules.yaml

```diff
@@ -0,0 +1,40 @@
+apiVersion: monitoring.coreos.com/v1
+kind: PrometheusRule
+metadata:
+  labels:
+    name: samples-operator-alerts
+  name: samples-operator-alerts
+  namespace: openshift-cluster-samples-operator
+spec:
+  groups:
+  - name: SamplesOperator
+    rules:
+    - alert: SampleImageStreamImportsFailing
+      # Testing showed that
+      # a) openshift_samples_failed_imagestream_import_total[2h] returned for each imagestream the count at each sample time
+      # b) max_over_time(openshift_samples_failed_imagestream_import_total[2h]) returned a max for each imagestream across all the sample times
+      # c) max(max_over_time(openshift_samples_failed_imagestream_import_total[2h])) returned a single element and max value across all the imagestreams
+      expr: max(max_over_time(openshift_samples_failed_imagestream_import_total[2h])) > 0
+      annotations:
+        message: |
+          Sample ImageStream imports are consistenly failing over 2 hours.
+    - alert: SamplesDegraded
+      expr: samples_operator_status_degraded == 1
+      annotations:
+        message: |
+          Samples could not be deployed and are in Degraded status.
+    - alert: SamplesInvalidConfig
+      expr: samples_operator_status_invalidconfig == 1
+      annotations:
+        message: |
+          Samples operator has been given an invalid configuration.
+    - alert: SamplesMissingSecret
+      expr: openshift_samples_status_invalidsecret{reason="missing-secret"} == 1
+      annotations:
+        message: |
+          Samples operator cannot find the samples pull secret in the openshift namespace.
+    - alert: SamplesMissingTBRCredential
+      expr: openshift_samples_status_invalidsecret{reason="missing-tbr-credential"} == 1
+      annotations:
+        message: |
+          Samples operator cannot find credentials for 'registry.redhat.io' in the samples pull secret in the openshift namespace.
```

Comment:
for a disconnected install there might not be a registry.redhat.io cred in the pullsecret....  i'd be wary of throwing this alert.

Reply:
good point

if disconnected, they should be setting the registry override in the samples config 

I'll augment the setting with testing that conditional first, and then update the message to add a clarification around that.

thoughts @bparees ?

Reply:
yeah i think that can work.